File: protocol/src/test/go/zfoogo/SimpleObject.go
package protocol

type SimpleObject struct {
	C int
    G bool
}

func (protocol SimpleObject) ProtocolId() int16 {
	return 104
}

func (protocol SimpleObject) write(buffer *ByteBuffer, packet any) {
	if packet == nil {
	    buffer.WriteInt(0)
		return
	}
	var message = packet.(*SimpleObject)
	buffer.WriteInt(-1)
    buffer.WriteInt(message.C)
    buffer.WriteBool(message.G)
}

func (protocol SimpleObject) read(buffer *ByteBuffer) any {
	var packet = new(SimpleObject)
	var length = buffer.ReadInt()
	if length == 0 {
		return packet
	}
	var beforeReadIndex = buffer.ReadOffset()
	var result0 = buffer.ReadInt()
    packet.C = result0
    var result1 = buffer.ReadBool()
    packet.G = result1
	if length > 0 {
        buffer.SetReadOffset(beforeReadIndex + length)
    }
	return packet
}

File: protocol/src/test/csharp/zfoocs/Packet/SimpleObject.cs
using System;
using System.Collections.Generic;
namespace zfoocs
{
    
    public class SimpleObject
    {
        public int c;
        public bool g;
    }

    public class SimpleObjectRegistration : IProtocolRegistration
    {
        public short ProtocolId()
        {
            return 104;
        }
    
        public void Write(ByteBuffer buffer, object packet)
        {
            if (packet == null)
            {
                buffer.WriteInt(0);
                return;
            }
            SimpleObject message = (SimpleObject) packet;
            buffer.WriteInt(-1);
            buffer.WriteInt(message.c);
            buffer.WriteBool(message.g);
        }
    
        public object Read(ByteBuffer buffer)
        {
            int length = buffer.ReadInt();
            if (length == 0)
            {
                return null;
            }
            int beforeReadIndex = buffer.ReadOffset();
            SimpleObject packet = new SimpleObject();
            int result0 = buffer.ReadInt();
            packet.c = result0;
            bool result1 = buffer.ReadBool();
            packet.g = result1;
            if (length > 0)
            {
                buffer.SetReadOffset(beforeReadIndex + length);
            }
            return packet;
        }
    }
}